File: MSSpotiAPI/src/main/java/com/ms/spotiapi/Models/Artist.java
package com.ms.spotiapi.Models;

import lombok.*;
import org.springframework.lang.Nullable;

import javax.persistence.*;
import java.io.Serializable;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name="ARTISTS")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Artist implements Serializable {
    @Id
    private String id;

    @PrimaryKeyJoinColumn
    public String name;

    @Nullable
    public int followers;

    @ManyToMany
    @JoinTable(
            name = "artists_genres",
                joinColumns = @JoinColumn(name="artist_name",referencedColumnName = "name"),
                inverseJoinColumns = @JoinColumn(name="genre_name",referencedColumnName = "name")
    )
    public Set<Genre> genres = new HashSet<>();

    public String popularity;

    public String uri;


}

File: LoadGenerator/Artist.py
import json


class Artist:

    def __init__(self, id, name, followers, genres, popularity, uri):
        self.id = id
        self.name = name
        self.followers = followers
        self.genres = genres
        self.popularity = popularity
        self.uri = uri

    def to_dict(self):
        genres = [genre.to_dict() for genre in self.genres]
        return {"id": self.id, "name": self.name, "followers": self.followers, "genres": genres,
                "popularity": self.popularity, "uri":self.uri
                }
